src/RTreeMap.cxx

```diff
@@ -2,39 +2,59 @@
 #include <ROOT/RBox.hxx>
 #include <ROOT/RText.hxx>
 #include <iomanip>
+#include <ROOT/RColumnElementBase.hxx>
 
-using Point = std::array<float, 2>;
-using Rect = std::array<Point, 2>;
+struct RRect {
+   RVec2 p1, p2;
+   RRect(const RVec2 &p1, const RVec2 &p2) : p1(p1), p2(p2) {}
+};
 
-#define TREEMAP_TEXTCOLOR RColor(255, 255, 255)
-constexpr float INDENTATION_OFFSET = 0.015f;
-constexpr float PAD_TEXT_OFFSET = 0.005f;
-constexpr float TEXT_SIZE_FACTOR = 0.01f;
+static RColor kTreemapTextColor = RColor(255, 255, 255);
+constexpr float kIndentationOffset = 0.015f;
+constexpr float kPadTextOffset = 0.005f;
+constexpr float kTextSizeFactor = 0.01f;
+const static std::vector<std::string> kUnits{"B", "KB", "MB", "GB", "TB", "PB", "EB"};
```

Comment:
Avoid global fields of non-trivial types, as they cause all sort of weird errors (sadly I had this happen to me recently due to simply declaring a global `static const std::string`).
In this case you should just declare this as `static const char *const kUnits[] = { ... }` or, equivalently `static constexpr const char *kUnits[] = { ... }`.
This makes the variable trivially constructible meaning it won't be subject to [SIOF](https://en.cppreference.com/w/cpp/language/siof.html) and in fact the compiler will likely put it directly in the .rodata section.